modeling-data-relationships/modeling-answers.md

```diff
@@ -0,0 +1,121 @@
+1)
+
+a)
+
+```
+SELECT guests.first_name, guests.last_name
+FROM guests
+LEFT JOIN bookings ON bookings.guest_id = guests.id
+WHERE bookings.guest_id IS NULL;
+```
+b) 
+```
+SELECT b.check_in, b.check_out, g.first_name, g.last_name, COUNT(*)
+FROM bookings AS b
+JOIN guests AS g ON g.id = b.guest_id
+GROUP BY b.check_in, b.check_out, g.first_name, g.last_name
+HAVING COUNT(*) = 2;
+```
+c)
+```
+SELECT b.room_id, g.first_name, g.last_name, COUNT(*)
+FROM guests AS g
+JOIN bookings AS b ON b.guest_id = g.id
+GROUP BY b.room_id, g.first_name, g.last_name
+HAVING COUNT(*) > 1;
+```
+
+d)
+```
+SELECT bookings.room_id, COUNT(DISTINCT guests.first_name) AS unique_guests
+FROM bookings 
+JOIN guests ON guests.id = bookings.guest_id
+GROUP BY bookings.room_id;
+```
+2)
+
+a)
+
+**What classes/entities do we need?**
+
+We definitely need a `students` entity, as well as a `course` entity in order to track the students in the school, as well as the courses give in the school. 
```

Comment:
but there is a physics101 every semester; how would you handle that?

Reply:
Urgh... 

I hadn't thought of that...

Hmmmmmm.....